includes/sanitizers/class-amp-meta-sanitizer.php

```diff
@@ -133,11 +133,81 @@ protected function ensure_charset_is_present() {
 	 * Always ensure we have a viewport tag.
 	 *
 	 * The viewport defaults to 'width=device-width', which is the bare minimum that AMP requires.
+	 * If there are @viewport style rules, these will have been moved into the content attribute of their own meta[name="viewport"] tags.
+	 * So this iterates over all of those tags, gets the content value,
+	 * and merges all of the valid values into a single meta tag.
 	 */
 	protected function ensure_viewport_is_present() {
 		if ( empty( $this->meta_tags[ self::TAG_VIEWPORT ] ) ) {
 			$this->meta_tags[ self::TAG_VIEWPORT ][] = $this->create_viewport_element( static::AMP_VIEWPORT );
+		} else {
+			$parsed_rules = [];
+
+			// Reverse the meta[name="viewport"] tags, so original meta[name="viewport"] tags have precedence over @viewport style rules.
+			foreach ( array_reverse( $this->meta_tags[ self::TAG_VIEWPORT ] ) as $meta_viewport ) {
```

Comment:
This prioritizes original `<meta name="viewport">` tags, not `@viewport` style rules that were added to a meta tag. Maybe this isn't what you had in mind, though.